src/slot_migrate.cc

```diff
@@ -568,6 +568,12 @@ Status SlotMigrate::MigrateOneKey(rocksdb::Slice key, std::string *restore_cmds)
     return Status(Status::cOK, "empty");;
   }
 
+  int64_t now;
+  rocksdb::Env::Default()->GetCurrentTime(&now);
+  if (metadata.expire <= now) {
```

Comment:
expire = 0 means never be expired, can use `metadata.Expired()`.

Reply:
Thank you for reminding me to check that `metadata.expire` must be greater than 0.
The reason why I did not use `metadata.Expired()` is that this method returns `false` when `metadata.expire == now`, which will make the source send a ttl of 0 to the destination, thus ruin the whole slot migration. I happened to meet such case before.

Reply:
Thank you for reminding me to check that `metadata.expire` must be greater than 0.
The reason why I did not use `metadata.Expired()` is that this method returns `false` when `metadata.expire == now`, which will make the source send a ttl of 0 to the destination, thus ruin the whole slot migration. I happened to meet such case before.

Reply:
I think there are two issues in this case:

1. Source node should never send known expired keys and we need to check whether is expired or not.
2. Target node should also skip expired keys instead of stoping the migration flow, since we can't guarantee the clock between source and target nodes are always the same and the network delay also may cause TTL = 0 problem.

Reply:
What do you think of:
1.  Source node still check whether the key has expired and skip it if it has. 
2. Source node employs `SET key value EXAT unix-time-seconds` (not supported yet as far as I know, but not hard to implement) instead of `SET key value EX ttl` to send the exact expire ttl during the migration, in order to address the clock syncing or network latency issue.
3. Target node checks the expire unix time when receiving `EXAT` and skip the key when it is behind the current system clock (Does the response "OK" looks reasonable?). The target node cannot respond an error since the source node will abort the migration upon receiving any error reply.

Reply:
Yes, exactly. I think it's ok to response ok when it's skipped.

Reply:
LGTM

Reply:
cool, thanks.